cpp/tests/binning/bin_tests.cu

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/binning/bin.hpp>
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <numeric>
+#include <thrust/device_ptr.h>
+#include <thrust/logical.h>
+#include <cudf/column/column.hpp>
+#include <cudf_test/type_lists.hpp>
+#include <thrust/execution_policy.h>
+
+using namespace cudf::test;
+
+namespace {
+
+// =============================================================================
+// ----- helper functions ------------------------------------------------------
+
+/// A simple struct to be used as a predicate for comparing a sequence to a given value encoded by this struct in algorithms.
+struct equal_value
+{
+    equal_value(unsigned int value)
```

Comment:
Please use the initialization style for constructor: `equal_value(... value) : m_value(value) {}`